spring-xd-spark-streaming-tests/src/test/java/org/springframework/xd/spark/streaming/SparkStreamingTests.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.xd.spark.streaming;
+
+import static org.hamcrest.core.IsEqual.*;
+import static org.junit.Assert.*;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Random;
+
+import org.junit.Test;
+
+import org.springframework.xd.shell.command.AbstractStreamIntegrationTest;
+import org.springframework.xd.shell.command.fixtures.HttpSource;
+import org.springframework.xd.shell.command.fixtures.XDMatchers;
+import org.springframework.xd.test.fixtures.FileSink;
+
+/**
+ * @author Ilayaperumal Gopinathan
+ */
+public class SparkStreamingTests extends AbstractStreamIntegrationTest {
+
+	static {
+		// Change the transport here to run the test against other message bus transports.
+		System.setProperty("XD_TRANSPORT", "local");
+	}
+
+	private static final String messageToPost = "foo foo foo";
```

Comment:
Not sure using a constant really helps readability, but if we do, should be UPPERCASE